dispatcher.go

```diff
@@ -107,6 +110,21 @@ func addObservingMiddleware(cfg Config, registry *pally.Registry, logger *zap.Lo
 	return cfg
 }
 
+func addIsRunningMiddleware(cfg Config, isRunning func() bool) Config {
+	outboundErr := pkgerrors.NotRunningOutboundError(cfg.Name)
+	inboundErr := pkgerrors.NotRunningInboundError(cfg.Name)
+
+	outboundIsRunningMiddleware := newIsRunningMiddleware(isRunning, outboundErr)
+	inboundIsRunningMiddleware := newIsRunningMiddleware(isRunning, inboundErr)
```

Comment:
nit: I think you can get away with a single middleware object (like observer)
if you let `newIsRunningMiddleware` instantiate the error objects.

```go
func newIsRunningMiddleware(serviceName string, isRunning func() bool) *isRunningMiddleware {
    return &isRunningMiddleware{
        isRunning: isRunning,
        inboundErr: pkgerrors.NotRunningInboundError(serviceName),
        outboundErr: pkgerrors.NotRunningOutboundError(serviceName),
    }
}
```

Reply:
Done

Reply:
Just a design decision - isRunning is agnostic to this being a dispatcher with a service name, it just takes a `func() bool` and and error if the function returns false. I could just name it `checkConditionMiddleware` but alas.